scrapi/processing/cassandra.py

```diff
@@ -77,8 +160,85 @@ def different(self, old, new):
         except Exception:
             return True  # If the document fails to load/compare for some reason, accept a new version
 
+    def documents(self, *sources):
+        sources = sources
+        q = DocumentModel.objects.timeout(500).allow_filtering().all().limit(1000)
+        querysets = (q.filter(source=source) for source in sources) if sources else [q]
+        for query in querysets:
+            page = try_n_times(5, list, query)
+            while len(page) > 0:
+                for doc in page:
+                    yield DocumentTuple(self.to_raw(doc), self.to_normalized(doc))
+                page = try_n_times(5, self.next_page, query, page)
+
+    def next_page(self, query, page):
+        return list(query.filter(docID__gt=page[-1].docID))
+
+    def to_raw(self, doc):
+
+        doc_items = dict(doc)
+
+        return RawDocument({
+            'doc': doc_items['doc'],
+            'docID': doc_items['docID'],
+            'source': doc_items['source'],
+            'filetype': doc_items['filetype'],
+            'timestamps': doc_items['timestamps'],
+            'versions': doc_items['versions']
+        }, validate=False, clean=False)
+
+    def to_normalized(self, doc):
+        # make the new dict actually contain real items
+        normed = {}
+        for key, value in dict(doc).items():
+            try:
+                normed[key] = json.loads(value)
+            except (ValueError, TypeError):
+                normed[key] = value
+
+        # Remove empty values and strip down to only normalized fields
+        do_not_include = ['docID', 'doc', 'filetype', 'timestamps', 'source']
+        for key in list(normed.keys()):
+            if not normed[key]:
+                del normed[key]
+
+        for key in list(normed.keys()):
+            if key in do_not_include:
+                del normed[key]
+
+        if normed.get('versions'):
+            normed['versions'] = list(map(str, normed['versions']))
+
+        # No datetime means the document wasn't normalized (probably wasn't on the approved list)
+        if normed.get('providerUpdatedDateTime'):
+            normed['providerUpdatedDateTime'] = normed['providerUpdatedDateTime'].isoformat()
```

Comment:
Might want to use the `scrapi.base.helpers.datetime_formatter` function instead, that way we can be sure it'll be in the expected form.